### PR TITLE
Adding Storage formulation with reserves 

### PR DESCRIPTION
**Pull request by sourabhdalvi**
cc. @kdayday

### CODE REVIEW

File: src/PowerSimulations.jl [hunk 1]
```diff
@@ -57,6 +57,7 @@ export HydroDispatchPumpedStoragewReservation
 ######## Renewable Formulations ########
 export BookKeeping
 export BookKeepingwReservation
+export BatteryDispatch
```
Comment (jd-lara) [inline src/PowerSimulations.jl:60]: BatteryAncialliryServices is a better descriptor.

**Comment by jd-lara** [comment]
For the record, this is the model being implemented: 

image

**Comment by sourabhdalvi** [comment]
> For the record, this is the model being implemented:
> 
> 

The formulation looks like this one, the one you list has some issues.
image

**Comment by jd-lara** [comment]
Final form: 

image

**Review by jd-lara — APPROVED** [review]
Great effort!

**Pull request merged**